src/main/java/org/selenium/pages/BasePage.java

```diff
@@ -0,0 +1,57 @@
+package org.selenium.pages;
+
+import com.google.gson.Gson;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.selenium.driver.Driver;
+import org.selenium.ojb.User;
+import org.selenium.utils.RandomGenerate;
+
+import java.io.*;
+import java.nio.file.NoSuchFileException;
+
+public abstract class BasePage {
+    protected WebDriver driver = Driver.getInstance().getWebDriver();
+    protected WebDriverWait wait = new WebDriverWait(driver, 30);
+    private static RandomGenerate random = new RandomGenerate();
+    private final Gson gson = new Gson();
+    protected User user = readUserFromGson(new File("src/main/resources/json/user.json"));
+
+    static  {
+        createJson();
+    }
+
```

Comment:
BasePage has some logic that doesn't belongs to it. It more looks like preparation steps
The best option is not to introduce such logic in the classes. If you need json to be created it is better to move itto beforeAll/beforeClass methods

Reply:
Fix it